implementations/c/ockam/transport/posix_socket/socket_tcp.c

```diff
@@ -255,7 +255,7 @@ ockam_error_t socket_tcp_read(void* ctx, uint8_t* buffer, size_t buffer_size, si
     bytes_read        = recv(p_socket->socket_fd, &recv_len, sizeof(uint16_t), 0);
     // Must convert from network order to native endianness
     p_transmission->transmit_length = ntohs(recv_len);
-    if (-1 == bytes_read) {
+    if (sizeof(uint16_t) != bytes_read) {
       error = TRANSPORT_ERROR_RECEIVE;
```

Comment:
Do we always expect a fixed amount to be read?

Reply:
AFAIK for tcp we first send size of the following data. Also, we pass sizeof(uint16_t) as a size of buffer where we want to receive data recv, so bytes_read cannot be more than that. Negative values are errors. Don't know if results from 0 to sizeof(uint16_t) could happen and how to handle them. @mikelodder7 @robinbudd thought?

Reply:
@mrinalwadhwa @mikelodder7 You'll never get 0 back from receive. If there's no data, you get -1 with errno set to EWOULD_BLOCK. If you get 1 byte, then the function fails.

And you'll never get more than sizeof(uint16_t) bytes b/c you've told TCP that that is your max buffer size.

So I think all cases were handled properly in the original code, although these changes also result in correct behavior.

Reply:
@robinbudd I wonder if bytes_read could be equal to 1 here...